### PR TITLE
Fix Resgen with Spaces in Path

### LINKED ISSUE #1280: RESX compilation can't handle spaces
Whenever I try to compile my project, and it includes a `*.resx` file, the compilation fails with weird errors if the full path to the RESX file contains a space. Move the project to a full path that has no spaces in it, and it compiles cleanly. Note that a project that does not contain any `resources` or `namedResources` always compiles correctly, even if its full path contains a space.

### PR DESCRIPTION
**Pull request by brthor**
Fixes #1280

### CODE REVIEW

File: test/dotnet-resgen.Tests/Microsoft.DotNet.Tools.Resgen.Tests.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        {
+            var spaceBufferDirectory = _root.CreateDirectory("space directory");
+            var testAppDir = spaceBufferDirectory.CreateDirectory("TestProjectWithResource");
+
+            CopyProjectToTempDir(Path.Combine(_testProjectsRoot, "TestProjectWithResource"), testAppDir);
+
+            var testProject = GetProjectPath(testAppDir);
+            var buildCommand = new BuildCommand(testProject);
+
+            var result = buildCommand.Execute().Should().Pass();
```
Comment (livarcocc) [inline test/dotnet-resgen.Tests/Microsoft.DotNet.Tools.Resgen.Tests.cs]: Why are you assigning here?

File: test/dotnet-resgen.Tests/Microsoft.DotNet.Tools.Resgen.Tests.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        private readonly TempDirectory _root;
+
+        public ResgenTests()
+        {
+            _testProjectsRoot = Path.Combine(AppContext.BaseDirectory, "TestAssets", "TestProjects");
+            _root = Temp.CreateDirectory();
+        }
+
+        [Fact]
+        public void Test_Build_Project_with_Resources_with_Space_in_Path()
```
Comment (livarcocc) [inline test/dotnet-resgen.Tests/Microsoft.DotNet.Tools.Resgen.Tests.cs]: Build_Project_with_Resources_with_Space_in_Path_Should_Succeed

File: test/dotnet-resgen.Tests/Microsoft.DotNet.Tools.Resgen.Tests.cs (hunk 1 above)
Comment (brthor) [inline test/dotnet-resgen.Tests/Microsoft.DotNet.Tools.Resgen.Tests.cs]: good point. Copy Pasta error :smile:

File: test/dotnet-resgen.Tests/Microsoft.DotNet.Tools.Resgen.Tests.cs (hunk 2 above)
Comment (brthor) [inline test/dotnet-resgen.Tests/Microsoft.DotNet.Tools.Resgen.Tests.cs]: :+1:

**Comment by livarcocc** [comment]
Squash commits. Other than that, LGTM.

**Comment by davidfowl** [comment]
Except for the fact the the CI is all red :smile:

**Comment by livarcocc** [comment]
@davidfowl Yes, but I figured that was implicit.

**Comment by brthor** [comment]
Of course will fix red before merging.

Would like for this to be 2 commits, one for test add and the other for product changes. Any opposition there?

File: test/dotnet-resgen.Tests/Microsoft.DotNet.Tools.Resgen.Tests.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        private readonly TempDirectory _root;
+
+        public ResgenTests()
+        {
+            _testProjectsRoot = Path.Combine(AppContext.BaseDirectory, "TestAssets", "TestProjects");
+            _root = Temp.CreateDirectory();
+        }
+
+        [Fact]
+        public void Test_Build_Project_with_Resources_with_Space_in_Path_Should_Succeed()
```
Comment (pakrym) [inline test/dotnet-resgen.Tests/Microsoft.DotNet.Tools.Resgen.Tests.cs]: Why are we running complete build just to test 2 lines of code? Can't this be tested with unit test?

Comment (brthor) [inline test/dotnet-resgen.Tests/Microsoft.DotNet.Tools.Resgen.Tests.cs]: Why do you say this is testing two lines of code? This is primarily testing that spaces in the path between the process chain involving resgen is handled correctly.

**Pull request merged**